velox/common/memory/MappedMemory.cpp

```diff
@@ -59,17 +60,57 @@ void MappedMemory::Allocation::findRun(
   VELOX_CHECK(false, "Seeking to an out of range offset in Allocation");
 }
 
-// Rounds 'value' to the next multiple of 'factor'.
-template <typename T, typename U>
-static inline T roundUp(T value, U factor) {
-  return (value + (factor - 1)) / factor * factor;
+MachinePageCount MappedMemory::ContiguousAllocation::numPages() const {
+  return bits::roundUp(size_, kPageSize) / kPageSize;
 }
 
 // static
 void MappedMemory::destroyTestOnly() {
   instance_ = nullptr;
 }
 
+std::string MappedMemory::toString() const {
+  return fmt::format("MappedMemory: Allocated pages {}", numAllocated());
+}
+
+MachinePageCount MappedMemory::allocationSize(
+    MachinePageCount numPages,
+    MachinePageCount minSizeClass,
+    std::array<int32_t, kMaxSizeClasses>& sizeIndices,
+    std::array<int32_t, kMaxSizeClasses>& sizeCounts,
+    int32_t& numSizes) const {
+  int32_t needed = numPages;
+  int32_t pagesToAlloc = 0;
+  numSizes = 0;
+  VELOX_CHECK_LE(
+      minSizeClass,
+      sizes_.back(),
+      "Requesting minimum size larger than largest size class");
+  for (int32_t sizeIndex = sizes_.size() - 1; sizeIndex >= 0; sizeIndex--) {
+    int32_t size = sizes_[sizeIndex];
+    bool isSmallest = sizeIndex == 0 || sizes_[sizeIndex - 1] < minSizeClass;
+    // If the size is less than 1/8 of the size from the next larger,
+    // use the next larger size.
+    if (size > (needed + (needed / 8)) && !isSmallest) {
+      continue;
+    }
+    int32_t numUnits = std::max(1, needed / size);
+    needed -= numUnits * size;
+    if (isSmallest && needed > 0) {
+      // If needed / size had a remainder, add one unit of smallest class.
```

Comment:
"add one unit of smallest class" - this comment is confusing as we are adding one unit of the same class as was used for other units.

Reply:
Repeating in case this comment got lost: "add one unit of smallest class" - this comment is confusing as we are adding one unit of the same class as was used for other units.